nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/CSVRecordReader.java

```diff
@@ -68,17 +68,19 @@ public Record nextRecord() throws IOException, MalformedRecordException {
             final Map<String, Object> rowValues = new HashMap<>(schema.getFieldCount());
 
             for (final RecordField recordField : schema.getFields()) {
-                String rawValue = csvRecord.get(recordField.getFieldName());
-                if (rawValue == null) {
+                String rawValue = null;
+                final String fieldName = recordField.getFieldName();
+                if (csvRecord.isSet(fieldName)) {
+                    rawValue = csvRecord.get(fieldName);
+                } else {
                     for (final String alias : recordField.getAliases()) {
-                        rawValue = csvRecord.get(alias);
-                        if (rawValue != null) {
+                        if (csvRecord.isSet(fieldName)) {
```

Comment:
@markap14 Shouldn't you be checking isSet(alias)? Since this is in the else, presumably isSet(fieldName) is still false?